packages/components/src/date-time/time.tsx

```diff
@@ -124,28 +120,30 @@ export function TimePicker( { is12Hour, currentTime, onChange } ) {
 	 * Function that sets the date state and calls the onChange with a new date.
 	 * The date is truncated at the minutes.
 	 *
-	 * @param {Object} newDate The date object.
+	 * @param {Moment} newDate The date object.
 	 */
-	function changeDate( newDate ) {
+	function changeDate( newDate: Moment ) {
 		setDate( newDate );
-		onChange( newDate.format( TIMEZONELESS_FORMAT ) );
+		onChange?.( newDate.format( TIMEZONELESS_FORMAT ) );
 	}
 
-	function update( name, value ) {
-		// If the 12-hour format is being used and the 'PM' period is selected, then
-		// the incoming value (which ranges 1-12) should be increased by 12 to match
-		// the expected 24-hour format.
-		let adjustedValue = value;
-		if ( name === 'hours' && is12Hour ) {
-			adjustedValue = from12hTo24h( value, am === 'PM' );
-		}
+	function update( name: 'date' | 'month' | 'year' | 'hours' | 'minutes' ) {
+		return ( value: number ) => {
```

Comment:
We usually try not to introduce runtime changes — is there a particular reason why we're doing so here?

Reply:
So if you go with `function update( name: string, value: number )` then you'll get an error because it can't be guaranteed that `name` will be a valid method in the moment object.

```js
		const newDate = date.clone()[ name ]( adjustedValue );
```

```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Moment'.
  No index signature with a parameter of type 'string' was found on type 'Moment'.
```

If you go with `function update( 	name: 'date' | 'month' | 'year' | 'hours' | 'minutes', value: number )` then you'll get an error because it can't be guaranteed that `input` will get back to you with a `name` that is one of `date`, `month`, `year`, `hours`, or `minutes`.

```js
				onUpdate={ update }
```

```
Type '(name: "date" | "month" | "year" | "hours" | "minutes", value: number) => void' is not assignable to type '(name: string, value: number) => void'.
  Types of parameters 'name' and 'name' are incompatible.
    Type 'string' is not assignable to type '"date" | "month" | "year" | "hours" | "minutes"'.
```

One solution could be to paramatize `UpdateOnBlurAsIntegerFieldProps`.

```js
export type UpdateOnBlurAsIntegerFieldProps< T extends string > = {
	name: T;
	onUpdate: ( name: T, value: number ) => void;
};
```

I think that is overly complex, though, so I went with changing `update` to be a partial function since that's a very common pattern and it's also how `updateAmPm` (immediately below this function) works.

Reply:
Thank you for the explanation — it makes sense to keep your changes